### PR TITLE
Removed old dynamism API from infer request

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Removed old dynamism API from Infer request internal

### Tickets:
 - 73952

### CODE REVIEW

File: src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp [hunk 1]
```diff
@@ -95,18 +109,19 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
     InputInfo::Ptr foundInput;
     DataPtr foundOutput;
     const bool isInput = findInputAndOutputBlobByName(name, foundInput, foundOutput);
```
Comment (ilya-lavrenov) [inline src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp:111]: do we need `isInput`?  I suppose now we can use `if (input)`

**Review by ilya-lavrenov — APPROVED** [review]

File: src/inference/dev_api/cpp_interfaces/interface/ie_iinfer_request_internal.hpp [hunk 2]
```diff
@@ -265,6 +265,9 @@ class INFERENCE_ENGINE_API_CLASS(IInferRequestInternal) : public std::enable_sha
      */
     bool findInputAndOutputBlobByName(const std::string& name, InputInfo::Ptr& foundInput, DataPtr& foundOutput) const;
 
+    std::shared_ptr<const ov::Node> findInputByNodeName(const std::string& name) const;
+    std::shared_ptr<const ov::Node> findOutputByNodeName(const std::string& name) const;
```
Comment (ilya-lavrenov) [inline src/inference/dev_api/cpp_interfaces/interface/ie_iinfer_request_internal.hpp:269]: please, add documentation for these new methods

File: src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp [hunk 3]
```diff
@@ -85,6 +83,22 @@ std::map<std::string, InferenceEngineProfileInfo> IInferRequestInternal::GetPerf
     IE_THROW(NotImplemented);
 }
 
+std::shared_ptr<const ov::Node> IInferRequestInternal::findInputByNodeName(const std::string& name) const {
+    for (const auto& input : GetInputs()) {
+        if (input->get_friendly_name() == name)
+            return input;
+    }
+    return nullptr;
```
Comment (mletavin) [inline src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp:91]: Shouldn't we throw an error if no input found with such a name?

Comment (ilyachur) [reply src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp:91]: No we shouldn't old API can work without inputs.

File: src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp (hunk 1 above)
Comment (ilyachur) [reply src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp:111]: In old API we can have empty input

**Pull request merged**